src/conf_mode/protocols_bgp.py

```diff
@@ -96,6 +96,11 @@ def verify(bgp):
                 # Check if the configure peer-group exists
                 if 'peer_group' in peer_config:
                     peer_group = peer_config['peer_group']
+
+                    # Check if peer-group exists
+                    if 'peer_group' not in asn_config:
```

Comment:
Statement will evaluate to `true` if no peer-group exists at all.

Line 104 should rather be changed to: `if 'peer_group' not in asn_config or peer_group not in asn_config['peer_group']:`

Reply:
Done